tools/misc/mapping_expander.py

```
@@ -0,0 +1,212 @@
+    try:
+        with open(path, "r", encoding="utf-8") as f:
+            exception_map = json.load(f)
+            if framework_exception:
+                print(f"ℹ️  Exceptions list for \"{framework_exception}\" loaded from JSON file \"{os.path.basename(path)}\"")
+            return exception_map
+    except Exception as e:
+        print(f"⚠️  [WARNING] Could not load exceptions JSON file: {e}")
+        return {}
+
+
+def expand_mappings(df_source, node_id_list, exception_map, reverse=False):
```

Pylint (3.3.7)

[refactor] Too many local variables (18/15)

(R0914)